src/librustdoc/html/render/mod.rs

```diff
@@ -103,27 +102,29 @@ crate fn ensure_trailing_slash(v: &str) -> impl fmt::Display + '_ {
 crate struct Context<'tcx> {
     /// Current hierarchy of components leading down to what's currently being
     /// rendered
-    crate current: Vec<String>,
+    current: Vec<String>,
     /// The current destination folder of where HTML artifacts should be placed.
     /// This changes as the context descends into the module hierarchy.
-    crate dst: PathBuf,
+    dst: PathBuf,
     /// A flag, which when `true`, will render pages which redirect to the
     /// real location of an item. This is used to allow external links to
     /// publicly reused items to redirect to the right location.
-    crate render_redirect_pages: bool,
-    /// The map used to ensure all generated 'id=' attributes are unique.
-    id_map: Rc<RefCell<IdMap>>,
-    /// Tracks section IDs for `Deref` targets so they match in both the main
-    /// body and the sidebar.
-    deref_id_map: Rc<RefCell<FxHashMap<DefId, String>>>,
-    crate shared: Arc<SharedContext<'tcx>>,
-    all: Rc<RefCell<AllTypes>>,
-    /// Storage for the errors produced while generating documentation so they
-    /// can be printed together at the end.
-    crate errors: Rc<Receiver<String>>,
-    crate cache: Rc<Cache>,
+    render_redirect_pages: bool,
+    /// Shared mutable state. We should probably redesign this.
+    shared: Rc<SharedContext<'tcx>>,
```

Comment:
I don't really understand this comment. What would the redesigned code look like?

Reply:
I was thinking that there might be a way to redesign it so that `SharedContext` is not a field in `Context` and instead `&mut SharedContext` is passed in wherever needed.

Reply:
I would open that as an issue instead of putting directly in the code. I don't know that this comment is actionable currently.

Reply:
Issue: #82381

Reply:
How's this: 9580b56

Reply:
It still seems a little silly to me to have that in the code, but I don't care enough to block on it.